### DISCUSSION TITLE
"public" folder

### DISCUSSION THREAD

**Discussion by nachoalcazar** [quasarframework/quasar#13830] [CLI - PWA mode]
Hello, 

Why when i launch "quasar dev" in cli I get access to all files of the public folder...

(example index.html. All folder "src" works OK)

![image](https://user-images.githubusercontent.com/94051561/176721331-698a6881-7a30-4f29-8010-01d2212c65bc.png)

but when I try to use PWA with "quasar dev -m pwa" the app doesn´t find any file of "public" folder:

(for example plugin pdfjs not work in this mode. Files not found)

![image](https://user-images.githubusercontent.com/94051561/176722980-4eb5e2e3-957b-433e-9fb1-16695d5d98da.png)

**Comment by Saeid-Za** [comment]
Hello There, I've tested this on fresh project created via quasar cli.
You can still access the index.html file on `localhost:8443/index.html` but due to root or `/` url being bound to internally created html file by quasar, we can't have a custom index.html overriding the default one that bootstraps vue and quasar.
Could you perhaps change the name of the served html file? a little explanation on the use-case would help a lot.

**Reply by nachoalcazar** [reply]
Thanks for the explanation. No problem to change my index.html, but... why the web app doesn't can access to other files from public in pwa mode?

For example:

![image](https://user-images.githubusercontent.com/94051561/177042272-c8b12fef-67ae-4d53-af3c-dc07c8725e58.png)

Folder icons: Left "quasar dev": right "quasar dev -m pwa"
![image](https://user-images.githubusercontent.com/94051561/177042280-a15a8d7c-8d9e-46cd-a1ae-849b71aa1108.png)

Extension @quasar/qpdfviewer left: "quasar dev -m pwa": right: "quasar dev"
![image](https://user-images.githubusercontent.com/94051561/177042296-c9272909-f710-451b-99a0-4b813c3c4445.png)

**Reply by Saeid-Za** [reply]
Oh, I see,
I Couldn't reproduce the issue on a fresh pwa project, could you please provide a minimal repository?
That way i can help more.

**Reply by nachoalcazar** [reply]
I've created a new project and only touch in quasar.config:

![image](https://user-images.githubusercontent.com/94051561/177054815-476913ee-21b8-482f-9fc9-776a878ea99d.png)

![image](https://user-images.githubusercontent.com/94051561/177054833-fd2a7f0d-1483-4e08-ae8d-7bbf49ded0d7.png)

and quasar mode add pwa

Then:

quasar dev:

![image](https://user-images.githubusercontent.com/94051561/177054857-93cbb243-6f5f-4c5b-b5ad-afebbf352e1e.png)

quasar dev -m pwa

![image](https://user-images.githubusercontent.com/94051561/177054916-7366d18d-9708-4d82-91da-144883510c21.png)

**Reply by Saeid-Za** [reply]
Hello. Sorry for late response.
I tried to reproduce this issue with the provided configuration. I've created a local CA and issued a certificate for `localhost` to serve the app in https mode (via `mkcert`).
Here's the changed parts of quasar.conf file, (only changed `vueRouterMode` and https part of devServer as specified.

```js

    // Full list of options: https://v2.quasar.dev/quasar-cli-vite/quasar-config-js#build
    build: {
      target: {
        browser: [ 'es2019', 'edge88', 'firefox78', 'chrome87', 'safari13.1' ],
        node: 'node16'
      },

      vueRouterMode: 'history', // available values: 'hash', 'history'
      // vueRouterBase,
...

    // Full list of options: https://v2.quasar.dev/quasar-cli-vite/quasar-config-js#devServer
    devServer: {
      https: {
        key: "E:/Contribution/localhost-key.pem",
        cert: "E:/Contribution/localhost.pem",
        ca: "E:/Contribution/rootCA.pem"
      },
      port: 8433,
      open: true // opens browser window automatically
    },

```

I still can access the `public` folder.

<img width="397" alt="sa" src="https://user-images.githubusercontent.com/65568529/179044807-aa11a568-88be-40a6-8fac-c29c712cfc29.png">

and here's my `quasar info` output.

```
Operating System - Windows_NT(10.0.22000) - win32/x64
NodeJs - 16.15.0

Global packages
  NPM - 8.5.5
  yarn - Not installed
  @quasar/cli - 1.3.2
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.7.5 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.0.5 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.14.3 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.37 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.1.2
  pinia - Not installed
  vuex - Not installed
  vite - 2.9.13 -- Native-ESM powered web dev build tool
  eslint - 8.19.0 -- An AST-based pattern checker for JavaScript.
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*
```

could you provide your `quasar info` output? maybe this is a platform specific problem.

**Reply by nachoalcazar** [reply]
Hello, 

have you add...

`quasar mode add pwa`

and launch...

`quasar dev -m pwa`

My quasar info:

Operating System - Windows_NT(10.0.19044) - win32/x64
NodeJs - 16.15.0

Global packages 
  NPM - 8.5.5
  yarn - 1.22.18
  @quasar/cli - 1.3.1
  @quasar/icongenie - Not installed
  cordova - 10.0.0 (cordova-lib@10.1.0)

Important local packages
  quasar - 2.7.4 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-webpack - 3.5.6 -- Quasar Framework App CLI with Webpack
  @quasar/extras - 1.14.2 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.37 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.0.16
  pinia - Not installed
  vuex - Not installed
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @babel/core - 7.18.6 -- Babel compiler core.
  webpack - 5.73.0 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 4.9.2 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - 6.5.3 -- A plugin for your Webpack build process, helping you generate a manifest of local files that workbox-sw should precache.
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  typescript - 4.5.5 -- TypeScript is a language for application scale JavaScript development
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - DESKTOP-1D3VNHC
  Ethernet - 192.168.1.75
  VirtualBox Host-Only Network - 192.168.56.1
